src/pylibs/db/tests/test_db_sql.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from sqlalchemy import MetaData
+from sqlalchemy.exc import IntegrityError
+from ..database_helper import DatabaseHelper
+from .constants import EXAMPLE_USER, EXAMPLE_CONTACT
+class TestDB(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.DB = DatabaseHelper('SQL', 'sqlite:///:memory:').database
+        cls.DB.users_table.create(cls.DB.engine)
+        cls.DB.contacts_table.create(cls.DB.engine)
+
+    def test_add_user_foo(self):
+        user=EXAMPLE_USER.copy()
+        #create a user with username foo
+        user['username']='foo'
+        user['accountid']=self.DB.generate_accountid()
```

Comment:
(med) Suggest creating separate tests for generate_accountid()

In this add_user() test, you end up also testing generate_accountid().  You should probably separate the tests by simply passing a literal accountid.

In tests for generate_accountid(), you can look into mocking the values for random.randint and verifying that generate_accountid() actually avoids duplicate account ids.